lib/puppet/provider/ec2_securitygroup/v2.rb

```diff
@@ -65,12 +65,28 @@ def self.format_ingress_rules(region, group, groups)
     group[:ip_permissions].each do |rule|
       addition = []
       rule.user_id_group_pairs.each do |security_group|
-        addition << prepare_ingress_rule_for_puppet(region, rule, groups, security_group)
+        addition << prepare_rule_for_puppet(region, rule, groups, security_group)
       end
       rule.ip_ranges.each do |cidr|
-        addition << prepare_ingress_rule_for_puppet(region, rule, groups, nil, cidr)
+        addition << prepare_rule_for_puppet(region, rule, groups, nil, cidr)
       end
-      addition << prepare_ingress_rule_for_puppet(region, rule, groups) if addition.empty?
+      addition << prepare_rule_for_puppet(region, rule, groups) if addition.empty?
+      rules << addition
+    end
+    rules.flatten.uniq.compact
+  end
+
+  def self.format_egress_rules(region, group, groups)
+    rules = []
+    group[:ip_permissions_egress].each do |rule|
```

Comment:
Does this and the method above only differ on the group they're working on? In that case it might make sense to change the method to 

    def self.format_rules(region, direction, group, groups)

and have the `direction` parameter indicate which one to use (`:ingress` or `:egress`)